pkg/apis/gcp/validation/backupbucket.go

```diff
@@ -31,3 +35,56 @@ func ValidateBackupBucketConfig(config *apisgcp.BackupBucketConfig, fldPath *fie
 
 	return allErrs
 }
+
+// ValidateBackupBucketConfigUpdate validates updates to the BackupBucketConfig.
+func ValidateBackupBucketConfigUpdate(oldConfig, newConfig *apisgcp.BackupBucketConfig, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs          = field.ErrorList{}
+		immutabilityPath = fldPath.Child("immutability")
+	)
+
+	if oldConfig.Immutability == nil || !oldConfig.Immutability.Locked {
+		return allErrs
+	}
+
+	if newConfig == nil || newConfig.Immutability == nil || *newConfig.Immutability == (apisgcp.ImmutableConfig{}) {
```

Comment:
Do we need the brackets?
```suggestion
	if newConfig == nil || newConfig.Immutability == nil || *newConfig.Immutability == apisgcp.ImmutableConfig{} {
```

Reply:
Looks like the linter is not happy without the brackets
```bash
$ make verify
...
> Check
Executing golangci-lint
pkg/apis/gcp/validation/backupbucket.go:1: : # github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp/validation [github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp/validation.test]
pkg/apis/gcp/validation/backupbucket.go:50:111: syntax error: unexpected { at end of statement
pkg/apis/gcp/validation/backupbucket.go:54:2: syntax error: non-declaration statement outside function body (typecheck)
// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
pkg/apis/gcp/validation/backupbucket.go:50:111: expected ';', found '{' (typecheck)
        if newConfig == nil || newConfig.Immutability == nil || *newConfig.Immutability == apisgcp.ImmutableConfig{} {
                                                                                                                     ^
pkg/apis/gcp/validation/backupbucket.go:54:2: expected declaration, found 'if' (typecheck)
        if !newConfig.Immutability.Locked {
        ^
pkg/apis/gcp/validation/backupbucket.go:85:2: expected declaration, found 'if' (typecheck)
        if !allowedGVKs.Has(credentialsRef.GroupVersionKind()) {
        ^
...
```